### PR TITLE
This adds a command line option to set data directory for the sdl2 platform

### PR DESCRIPTION
**Pull request by IMbackK**
this fixes https://github.com/XProger/OpenLara/issues/400 by adding a command line option (-d) to set the data directory.

this pr also fixes the sdl2 makefile in e55bf112f24976b09e6466751ea4bf9062c60f04, which was previously non functional.

side note: while fixing the makefile i noticed that -Wall creates a lot of warnings, a cursory glance at some of these reveals some that strongly look like real bugs, please consider adding -Wall to your private builds and fixing the warnings.

### CODE REVIEW

File: src/platform/sdl2/main.cpp [hunk 1]
```diff
@@ -463,7 +463,53 @@ void inputUpdate() {
     }
 }
 
+void print_help(int argc, char **argv) {
+    
+    printf("%s [OPTION]\nA open source re-implementation of the classic Tomb Raider engine.\n",
```
Comment (XProger) [inline src/platform/sdl2/main.cpp:468]: argv[0]?

**Review by XProger — COMMENTED** [review]
@vanfanel fyi

File: src/platform/sdl2/main.cpp (hunk 1 above)
Comment (IMbackK) [reply src/platform/sdl2/main.cpp:468]: Not sure what the objection is here, argv[0] contains the name of the binary with its calling path ie ./openlara or whatever the binary is called.

File: src/platform/sdl2/main.cpp [hunk 2]
```diff
@@ -353,10 +353,10 @@ void inputUpdate() {
             }
 
             case SDL_KEYUP: {
```
[the remark this answers is not in the document]
Comment (IMbackK) [reply src/platform/sdl2/main.cpp:354]: fixed, also some more other white space problems fixed

**Pull request merged**